map/game_map.py

```diff
@@ -0,0 +1,58 @@
+from collections import Counter
+from math import floor
+import numpy
+from scipy import stats
+
+from tcod.console import Console
+
+from map import tile_types
+from render_functions import (
+    screen_width, 
+    screen_height,
+    map_width, 
+    map_height, 
+    map_view_width, 
+    map_view_height
+)
+
+
+class GameMap:
+    def __init__(self, width: int, height: int):
+        self.width, self.height = width, height
+        self.tiles = numpy.full((width, height), fill_value=tile_types.ground, order="F")
+        self.zoomed_out_tiles = numpy.full((floor(self.width/3), floor(self.height/3)), fill_value=tile_types.ground, order="F")
+        
+        self.tiles[29:33, 29] = tile_types.wall
+        self.tiles[29, 29:33] = tile_types.wall
+        self.tiles[33, 29:33] = tile_types.wall
+
+        
+
+    def in_bounds(self, x: int, y: int) -> bool:
+        """Return True if x and y are inside of the bounds of this map."""
+        return 0 <= x < self.width and 0 <= y < self.height
+
+
+    def average_of_tiles(self, tiles):
+        tile = numpy.where(tiles["tile_type"] > 99) #not working
+        if tile[0].size > 0:
+            return tiles[tile[0][0], tile[1][0]]
+        else:
+            #would prefer to return the mode tile type, but if above we are (eventually) checking for
+            #"important" tiles, then taking the middle tile should be harmless
```

Comment:
Note to self: Not exactly harmless. I want this function to be reusable. Say we want to make a minimap in the HUD that zooms out even more.. We want to be able to take the most common tile